modules/cache/cache.go

```diff
@@ -34,25 +35,38 @@ func NewContext() error {
 		if conn, err = newCache(setting.CacheService.Cache); err != nil {
 			return err
 		}
-		const testKey = "__gitea_cache_test"
-		const testVal = "test-value"
-		if err = conn.Put(testKey, testVal, 10); err != nil {
+		err = Ping()
+		if err != nil {
 			return err
 		}
-		val := conn.Get(testKey)
-		if valStr, ok := val.(string); !ok || valStr != testVal {
-			// If the cache is full, the Get may not read the expected value stored by Put.
-			// Since we have checked that Put can success, so we just show a warning here, do not return an error to panic.
-			log.Warn("cache (adapter:%s, config:%s) doesn't seem to work correctly, set test value '%v' but get '%v'",
-				setting.CacheService.Cache.Adapter, setting.CacheService.Cache.Conn,
-				testVal, val,
-			)
-		}
 	}
 
 	return err
 }
 
+// Ping checks if the cache service works or not, it not, it returns an error
+func Ping() error {
+	if conn == nil {
+		return errors.New("cache not available")
+	}
+	var err error
+	const testKey = "__gitea_cache_test"
+	const testVal = "test-value"
+	if err = conn.Put(testKey, testVal, 10); err != nil {
+		return err
+	}
+	val := conn.Get(testKey)
+	if valStr, ok := val.(string); !ok || valStr != testVal {
+		// If the cache is full, the Get may not read the expected value stored by Put.
+		// Since we have checked that Put can success, so we just show a warning here, do not return an error to panic.
+		log.Warn("cache (adapter:%s, config:%s) doesn't seem to work correctly, set test value '%v' but get '%v'",
+			setting.CacheService.Cache.Adapter, setting.CacheService.Cache.Conn,
+			testVal, val,
+		)
+	}
+	return nil
+}
+
```

Comment:
Redis has a [PING](https://redis.io/commands/PING) command, but I guess we can't use it in this abstraction, right?

Reply:
I would prefere that too ... but this will need a pull to https://gitea.com/go-chi/cache/src/branch/master/cache.go#L24 so it can be passed down to it's implementations ... and then this "cache ping" hack can be used there if the underlying implementation do not have a ping ...

Reply:
Ok, maybe something to do later to not complicate this here further.

Reply:
Should we open an issue about it on https://gitea.com/go-chi/cache?

Reply:
yes !

Reply:
just for mark: https://gitea.com/go-chi/cache/issues/3